### DISCUSSION TITLE
How to check the noise budget?

### DISCUSSION THREAD

**Discussion by fedejinich** [tuneinsight/lattigo#412] [Q&A] [answered]
Hi, how can I get the remaining noise budget of a ciphertext? I want to see the way noise grows during the execution, and I want to check how much noise is left until I corrupt the ciphertext. Is there any util for that? 
Thanks

**Comment by Pro7ech** [comment, marked as answer]
Hi @fedejinich, I believe you are referring to the BFV/BGV schemes. The only way to check the noise budget is to decrypt and check the norm of the error. See the tests in the BFV/BGV schemes on how to do it:

https://github.com/tuneinsight/lattigo/blob/edde68e6a316c47cc9e035afedc80cc9ec5a622b/schemes/bgv/bgv_test.go#L159

**Reply by fedejinich** [reply]
thanks!
